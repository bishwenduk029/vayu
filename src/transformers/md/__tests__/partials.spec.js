import { compile } from "../index";

const FIXTURES_ROOT_FOLDER =
  "/Users/kundb/project/vayu/src/transformers/md/__tests__/fixtures";

test("should parse partials together into a single object graph", async () => {
  const frontMatter = await compile(`${FIXTURES_ROOT_FOLDER}/test.md`, null, {
    contentFolder: FIXTURES_ROOT_FOLDER,
  });
  expect(frontMatter).toMatchSnapshot();
});

test("should throw an error if the partial is not having any key", async () => {
  try {
    await compile(`${FIXTURES_ROOT_FOLDER}/errorContent.md`, null, {
      contentFolder: FIXTURES_ROOT_FOLDER,
    });
  } catch (error) {
    expect(error).toMatchSnapshot();
  }
});

test("should throw an error if the partial does not exist", async () => {
  try {
    await compile(`${FIXTURES_ROOT_FOLDER}/missingPartialContent.md`, null, {
      contentFolder: FIXTURES_ROOT_FOLDER,
    });
  } catch (error) {
    expect(error).toMatchSnapshot();
  }
});

test("should parse aggregations correctly", async () => {
  const frontMatter = await compile(
    `${FIXTURES_ROOT_FOLDER}/test_aggr.md`,
    null,
    {
      contentFolder: FIXTURES_ROOT_FOLDER,
    }
  );
  expect(frontMatter).toMatchSnapshot();
});

test("should parse simple markdowns without collections correctly", async () => {
  const frontMatter = await compile(
    `${FIXTURES_ROOT_FOLDER}/test_simple.md`,
    null,
    {
      contentFolder: FIXTURES_ROOT_FOLDER,
    }
  );
  expect(frontMatter).toMatchSnapshot();
});
