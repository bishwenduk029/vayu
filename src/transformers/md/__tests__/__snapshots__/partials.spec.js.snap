// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should parse aggregations correctly 1`] = `
Object {
  "content": "",
  "data": Object {
    "aggregator": Object {
      "Getting Started": Array [
        Object {
          "content": "",
          "data": Object {
            "key": "section10",
            "key1": "value1",
            "key2": "value2",
          },
        },
        Object {
          "content": "",
          "data": Object {
            "key": "section9",
            "key1": "value1",
            "key2": "value2",
          },
        },
      ],
    },
    "key1": "value1",
    "key2": "value2",
    "partials": Array [
      "section2.md",
      "section3.md",
    ],
    "section2": Object {
      "content": "",
      "data": Object {
        "key": "section2",
        "key1": "value1",
        "key2": "value2",
      },
    },
    "section3": Object {
      "content": "",
      "data": Object {
        "key": "section3",
        "key1": "value1",
        "key2": "value2",
        "section4": "./section4.md",
      },
    },
  },
}
`;

exports[`should parse partials together into a single object graph 1`] = `
Object {
  "content": "",
  "data": Object {
    "key1": "value1",
    "key2": "value2",
    "partials": Array [
      "section2.md",
      "section3.md",
    ],
    "section2": Object {
      "content": "",
      "data": Object {
        "key": "section2",
        "key1": "value1",
        "key2": "value2",
      },
    },
    "section3": Object {
      "content": "",
      "data": Object {
        "key": "section3",
        "key1": "value1",
        "key2": "value2",
        "section4": "./section4.md",
      },
    },
  },
}
`;

exports[`should parse simple markdowns without collections correctly 1`] = `
Object {
  "content": "",
  "data": Object {
    "key1": "value1",
    "key2": "value2",
  },
}
`;

exports[`should throw an error if the partial does not exist 1`] = `[Error: ENOENT: no such file or directory, open '/Users/kundb/project/vayu/src/transformers/md/__tests__/fixtures/section99.md']`;

exports[`should throw an error if the partial is not having any key 1`] = `[Error: FrontMatter of the file /Users/kundb/project/vayu/src/transformers/md/__tests__/fixtures/section5.md has a syntax error. If it is an abstract/partial content file then it should have a key]`;
